maintenance/measure_tile_performance/measure_tile_performance.py

```diff
@@ -7,97 +7,135 @@
-    cache_only=False)
-    # Convert string to Python dict 
-    query_dic = json.loads(response) 
-    runtime_duration = round(float(query_dic['runtime']), 2)
-    ran_at_utc = pytz.utc.localize(datetime.fromisoformat(query_dic['ran_at'][:-1]))
-    ran_at = ran_at_utc.astimezone(pytz.timezone('America/Vancouver')).isoformat()
-    query_list.append(runtime_duration)
     if args.file:
-        with open(filename, 'a', encoding='UTF8') as f:
+        with open(filename, 'w', encoding='UTF8') as f:
             writer = csv.writer(f)
-            # write the data
-            writer.writerow([slug_input, ran_at, runtime_duration])
-    
-    # adding 5 min time delay between subsequent queries to keep looker open for other tasks
-    if i != times:
-        time.sleep(sleep_timer)
-
-    i += 1
-
-if args.file:
-    print("Filename =", filename)
-    f.close()
-    
-#Print for Console 
-
-print("RunTimes =", query_list)
-minimum = min(query_list)
-print("Min =", minimum)
-maximum = max(query_list)
-print("Max =", maximum)
-sum = sum(query_list)
-length = len(query_list)
-average = round(sum/length, 2)
-print("Avg =", average)
+            # write the header
+            writer.writerow(header)
+
+    # run query as per user input
+
+    i = 1
+    while i <= times :
+        response = sdk.run_query(
+        query_id=query,
+        result_format="json_detail",
+        cache=False,
+        cache_only=False)
+        # Convert string to Python dict 
+        query_dic = json.loads(response) 
+        runtime_duration = round(float(query_dic['runtime']), 2)
+        ran_at_utc = pytz.utc.localize(datetime.fromisoformat(query_dic['ran_at'][:-1]))
+        ran_at = ran_at_utc.astimezone(pytz.timezone('America/Vancouver')).isoformat()
+        query_list.append(runtime_duration)
+        if args.file:
+            with open(filename, 'a', encoding='UTF8') as f:
+                writer = csv.writer(f)
+                # write the data
+                writer.writerow([slug_input, ran_at, runtime_duration])
+        
+        # adding 5 min time delay between subsequent queries to keep looker open for other tasks
+        if i != times:
+            time.sleep(sleep_timer)
+
+        i += 1
 
-    
+    if args.file:
+        print("Filename =", filename)
+        f.close()
+        
+    #Print for Console 
+    print("RunTimes =", query_list)
+    minimum = min(query_list)
+    print("Min =", minimum)
+    maximum = max(query_list)
+    print("Max =", maximum)
+    sum_of_list = sum(query_list)
+    length = len(query_list)
+    average = round(sum_of_list/length, 2)
+    print("Avg =", average)
+    #if query runs are more than 1, then calculate standard deviation
+    if times > 1:
+        standard_dev = round(statistics.stdev(query_list), 2)
+        print("Standard Deviation =", standard_dev)
+    with conn:
+        with conn.cursor() as curs:
+            curs.execute("ALTER USER looker SET enable_result_cache_for_session TO on;")
+
```

Comment:
And this is being run later than it could be. There is unnecessary delay between `off` and `on` for the looker user cache enabling.